doc: introduce translator comment policy documentation

jarolrod opened the pull request:
This serves to document our policy of requiring translator comments for any user-facing string as well as provide guidance for how to construct these translator comments.

Questions for reviewer:
- Should this be a general localization policy document, which includes the translator comment policy as a part of it?
  - There could be benefit in documenting our full list of policies around localizations as what not to use and what is ok to use, etc ...
- Concept N/ACK on general direction of the information contained in the doc?

[render](https://github.com/bitcoin-core/gui-qml/blob/a4fd76f47f290e2cfb21d040383409d9ac30be81/src/qml/doc/translator-comments.md)

promag commented on src/qml/doc/translator-comments.md:
````diff
@@ -0,0 +1,60 @@
+Below, lines 4 and 5 are examples of new user-facing strings marked for
+translation without translator comments:
+
+```c++
````
qml, same below

promag commented on src/qml/doc/translator-comments.md:
````diff
@@ -0,0 +1,60 @@
+translation without translator comments:
+
+```c++
+1.  OptionButton {
````
Why line numbers? My suggestion is to drop them and above say "Below, `text` and `description` values are examples of new user-facing ..."

promag commented on src/qml/doc/translator-comments.md:
````diff
@@ -0,0 +1,60 @@
+9.     //: If there is no clear translation for "cellular data", use the translation for "mobile data"
+10.    description: qsTr("Loads quickly when on wi-fi and pauses when on cellular data.")
+11.  }
+```
````
Note that `//:` has special meaning, maybe point to https://doc.qt.io/qt-5/qtquick-internationalization.html#2-add-context-for-the-translator.

jarolrod replied:
Done, but I didn't include the link because I don't want a translator getting confused by the other variations of this character that we won't be using.

promag reviewed (commented):
Concept ACK, very nice!

hebasto reviewed (commented):
Concept ACK. Un-draft?

jarolrod commented:
Updated from 2fbd26a ->  a4fd76f ([pr81.01](https://github.com/jarolrod/gui-qml/tree/pr81.01) -> [pr81.02](https://github.com/jarolrod/gui-qml/tree/pr81.02))

Changes: addressed review comments and made various changes. 

@GBKS @shaavan would this documentation help you in constructing translator comments? What can be improved?

shaavan commented on src/qml/doc/translator-comments.md:
```diff
@@ -0,0 +1,86 @@
+
+Translator comments are required for all user-facing strings. A PR introducing
+new user-facing strings must also submit the accompanying translator comment.
+To prevent this requirement from halting development, the PR author can specify
+that they would like the introduction of the comment to be picked up in a
+follow-up PR.
```
I understand that this might be part of official guidelines and must be mentioned here. But I was just curious, wouldn’t this guideline indirectly motivate a contributor to laze out on translator comments and put it for future PR, potentially increasing translation burden on the complete project.

shaavan commented on src/qml/doc/translator-comments.md:
```diff
@@ -0,0 +1,86 @@
+If there is no clear translation for "cellular data", use the translation for
+"mobile data".
+
+Now take these texts, split them into separate sentences, and append `//:`
+to them. Place the text above its respective translatable string:
```
Can’t we use `/*: ... */` for multiline comments, as we used to do in the older GUI?

Also, what do you think about adding a little documentation for standardizing the spacing of comments? That is a classic beginner’s mistake from which I also used to suffer from :)

shaavan reviewed (commented):
Concept ACK

A solid Documentation! Great work, as always, @jarolrod! This documentation will definitely help me clarify how to write translator comments to have an easier time translating.

There are some minor things I would like to clarify, though.

GBKS commented:
Concept ACK. This is great. I do not have any feedback on the implementation details (you have more experience with this) and will follow whatever formatting style you will decide on.

hebasto commented:
I've restarted a CI build.

hebasto reviewed (approved):
ACK a4fd76f47f290e2cfb21d040383409d9ac30be81
